eclipse/src/saros/ui/wizards/pages/EnterProjectNamePage.java

```diff
@@ -371,11 +382,27 @@ private void preselectProjectNames() {
 
       if (project != null) continue;
 
-      String projectNameProposal =
-          findProjectNameProposal(
-              remoteProjectMapping.get(projectID), reservedProjectNames.toArray(new String[0]));
+      final String remoteProjectName = remoteProjectIdToNameMapping.get(projectID);
```

Comment:
Why not separating the into two private methods (corresponding to the two steps in the javadoc) in order to improve the readability?

Reply:
I guess that might require a little bit of more work. I will look into it if it is possible to split this without adding more checks and corrections.

Reply:
So after thinking about it for a while. Does it really make sense to split this ? I know that methods should not be that long or complex but the problem I have all the time is how to split methods ... The problem is in general the mutating state of such methods. If I split method A into multiple sub methods I have to ensure that each method called in another context does always produce the same result. In praxis (as I have seen it) you are ending up with a class consisting of so much methods that over time nobody has a clue what each method really does without reading the code.

The problem on this method is the second part. If I split this I am losing the project names that are already assigned through the first part. To get around of this issues I would have to query the enabled state which can also be mutated by any other method or logic. 

This class is going to be deleted / replaced anyway. I am still waiting to see the HTML equivalent of this class. Not one student have every tried to implement this wizard yet in the HTML GUI.